app/static/src/store/graphs/getters.ts

```diff
@@ -33,9 +33,11 @@ export const getters: GraphsGetters & GetterTree<GraphsState, AppState> = {
     },
     [GraphsGetter.legendWidth]: (_, graphsGetters): number => {
         // Heuristic for setting graph legend width based on string length of longest variable name
-        const longestVar = graphsGetters[GraphsGetter.allSelectedVariables].sort((a: string, b: string) =>
-            a.length < b.length ? 1 : -1
-        )[0];
-        return longestVar.length * LEGEND_WIDTH_PER_CHAR + LEGEND_LINE_PADDING;
+        let maxVariableLength = 0;
+        const variables = graphsGetters[GraphsGetter.allSelectedVariables];
+        for (let i = 0; i < variables.length; i++) {
+            maxVariableLength = Math.max(maxVariableLength, variables[i].length);
+        }
+        return maxVariableLength * LEGEND_WIDTH_PER_CHAR + LEGEND_LINE_PADDING;
```

Comment:
This the change where the sort was reordering the result of the getter - because the getter result is cached and reused when some drop-down uses it? Which drop down was that?

Reply:
yep exactly its being reused but sort also mutated the array in place, it was setting the wrong order in the select in `LinkData.vue`